shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJob.java

```diff
@@ -53,7 +53,7 @@ public void execute(final ShardingContext shardingContext) {
         } catch (final RuntimeException ex) {
             // CHECKSTYLE:ON
             log.error("job prepare failed, {}-{}", shardingContext.getJobName(), shardingContext.getShardingItem());
-            jobContext.close();
+            RuleAlteredJobWorker.getPIPELINE_JOB_EXECUTOR().stopJob(shardingContext.getJobName());
```

Comment:
If `jobContext.close();` removed, looks it might cause resource leak.

`RuleAlteredJobWorker.getPIPELINE_JOB_EXECUTOR().` could be replaced by `RuleAlteredJobSchedulerCenter.stop`.

Reply:
RuleAlteredJobSchedulerCenter.stop method will close jobContext.close()